guides/source/sign_up_and_settings.md

```diff
@@ -0,0 +1,1652 @@
+
+We'll create a namespace called "Settings" to separate out the user and store settings from the rest of our application.
+
+In `config/routes.rb` we can add the Settings namespace along with a resource for editing passwords:
+
+```ruby
+namespace :settings do
+  resource :password, only: [:show, :update]
+end
+```
+
+This will generate a route for `/settings/password` for editing the current user's password and is separate from the password resets routes at `/password`.
+
+### Adding the Namespaced Passwords Controller & View
+
+Namespaces also move controllers into a matching module in Ruby. This controller will be in a `settings` folder to match the namespace.
+
+Let's create the controller at `app/controllers/settings/passwords_controller.rb` and start with the `show` action.
+
+```ruby
+class Settings::PasswordsController < ApplicationController
+  def show
+  end
+end
+```
+
+Views also move to a `settings` folder so we can create `app/views/settings/passwords/show.html.erb` for this action.
+
+```erb
+<h1>Password</h1>
+
+<%= form_with model: Current.user, url: settings_password_path do |form| %>
+  <% if form.object.errors.any? %>
+    <div><%= form.object.errors.full_messages.first %></div>
+  <% end %>
+
+  <div>
+    <%= form.label :password_challenge %>
+    <%= form.password_field :password_challenge, required: true, autocomplete: "current-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password %>
+    <%= form.password_field :password, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password_confirmation %>
+    <%= form.password_field :password_confirmation, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.submit "Update password" %>
+  </div>
+<% end %>
+```
+
+We've set the `url:` argument to ensure the form submits to our namespaced route and is processed by the `Settings::PasswordsController`. Passing `model: Current.user` also tells `form_with` to submit a `PATCH` request to process the form with the `update` action.
+
+### Safely Updating Passwords
+
+Let's add the `update` action to the controller now.
+
+```ruby#5-17
+class Settings::PasswordsController < Settings::BaseController
+  def show
+  end
+
+  def update
+    if Current.user.update(password_params)
+      redirect_to settings_profile_path, status: :see_other, notice: "Your password has been updated."
+    else
+      render :show, status: :unprocessable_entity
+    end
+  end
+
+  private
+    def password_params
+      params.expect(user: [ :password, :password_confirmation, :password_challenge ]).with_defaults(password_challenge: "")
+    end
+end
+```
+
+For security, we need to ensure that only the user is the only one who can update their password. The `has_secure_password` method in our `User` model provides this attribute. If `password_challenge` is not `nil`, it will validate the password challenge against the password in the database to confirm it matches.
```

Comment:
```suggestion
For security, we need to ensure that the user is the only one who can update their password. The `has_secure_password` method in our `User` model provides this attribute. If `password_challenge` is not `nil`, it will validate the password challenge against the password in the database to confirm it matches.
```

Removed a second 'only'.